### ISSUE TITLE
esm/commonjs import issue

### ISSUE THREAD

**Issue by bjornsnoen** [CyanSalt/vite-plugin-legacy-swc#1]
Trying to import this legacy plugin like so
```typescript
import { defineConfig } from "vite";
import react from "@vitejs/plugin-react-swc";
import legacy from "vite-plugin-legacy-swc";

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [react(), legacy()],
});
```
causes the following error:
```bash
~/src/vite-project  master ✔                                                                                                                                                                          ◒  
▶ yarn build                                             
yarn run v1.22.19
$ tsc && vite build
failed to load config from /home/bjorn/src/vite-project/vite.config.ts
error during build:
TypeError: legacy is not a function
    at file:///home/bjorn/src/vite-project/vite.config.ts.timestamp-1676562433105.mjs:6:22
    at ModuleJob.run (node:internal/modules/esm/module_job:193:25)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:530:24)
    at async loadConfigFromBundledFile (file:///home/bjorn/src/vite-project/node_modules/vite/dist/node/chunks/dep-3007b26d.js:62302:21)
    at async loadConfigFromFile (file:///home/bjorn/src/vite-project/node_modules/vite/dist/node/chunks/dep-3007b26d.js:62187:28)
    at async resolveConfig (file:///home/bjorn/src/vite-project/node_modules/vite/dist/node/chunks/dep-3007b26d.js:61808:28)
    at async build (file:///home/bjorn/src/vite-project/node_modules/vite/dist/node/chunks/dep-3007b26d.js:44492:20)
    at async CAC.<anonymous> (file:///home/bjorn/src/vite-project/node_modules/vite/dist/node/cli.js:813:9)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

In order to make it work, you have to do this
```typescript
import legacyPkg from 'vite-plugin-legacy-swc'
// @ts-expect-error the package type definition is wrong
const legacy = legacyPkg.default as typeof legacyPkg
```

Steps to recreate:
```bash
$ yarn create vite
$ # Select React, Typescript + swc
$ yarn add -D vite-plugin-legacy-swc
$ # Edit vite.config.ts to match the first one in this issue
$ yarn build
```

Node version: v18.12.0

**Comment by CyanSalt** [comment]
Thank you so much for your feedback!

I've just released version [0.3.0](https://github.com/CyanSalt/vite-plugin-legacy-swc/releases/tag/v0.3.0) to try and support native ESM, which I've never used before, and I can only guarantee that it still supports CommonJS well!

**Comment by bjornsnoen** [comment]
Great, version 0.3 seems to solve it and is working fine, thanks for the quick response!
